src/BenchmarksClient/Startup.cs

```diff
@@ -100,72 +100,89 @@ public static int Main(string[] args)
         private static async Task ProcessJobs(CancellationToken cancellationToken)
         {
             IWorker worker = null;
-
+            var allJobs = _jobs.GetAll();
+            // Dequeue the first job. We will only pass jobs that have
+            // the same SpanId to the current worker.
+            var job = allJobs.FirstOrDefault();
             while (!cancellationToken.IsCancellationRequested)
             {
-                var allJobs = _jobs.GetAll();
-                var job = allJobs.FirstOrDefault();
                 if (job != null)
                 {
-                    if (job.State == ClientState.Waiting)
+                    if (job.State == ClientJobState.Waiting)
                     {
                         Log($"Starting '{job.Client}' worker");
-                        job.State = ClientState.Starting;
+                        job.State = ClientJobState.Starting;
 
                         try
                         {
-                            worker = WorkerFactory.CreateWorker(job);
+                            if (worker == null)
+                            {
+                                worker = WorkerFactory.CreateWorker(job);
+                            }
 
                             if (worker == null)
```

Comment:
There should be an `else` here and check the if `SpanId` matches the current `worker`. If not, `continue` skip the job.

Reply:
https://github.com/aspnet/benchmarks/pull/448/files#diff-d375e6377c60b0a87d2709204a22dd73R174
This should handle that logic. The first job we select is just the first one queued but after that we only select jobs that have the same SpanId